_source/_posts/2022-10-01-spring-vault.md

```diff
@@ -0,0 +1,639 @@
+
+Enable a file audit device to watch the interactions with Vault:
+
+```shell
+vault audit enable file file_path=/vault/logs/vault_audit.log
+```
+
+You should see a success message. Now store the Okta secrets for the `vault-demo-app`:
+
+```shell
+vault kv put secret/vault-demo-app,dev \
+okta.oauth2.clientId="{yourClientId}" \
+okta.oauth2.clientSecret="{yourClientSecret}" \
+okta.oauth2.issuer="{yourIssuerURI}"
+```
+```shell
+vault kv get secret/vault-demo-app,dev
+```
+
+As illustrated above, key-value pairs are stored with `kv put` command, and you can check the values with the `kv get` vault command.
+
+Check `vault_audit.log` in your specified `{hostPath}` directory. Operations are logged in JSON format by default, with sensitive information hashed:
+
+```json
+{
+   "time":"2022-09-09T22:09:14.886126069Z",
+   "type":"request",
+   "auth":{
+      "client_token":"hmac-sha256:2d8d7aeded539495117218a14ef9ae9f0f525eb2f6078be25dfe8e329ceb3d3f",
+      "accessor":"hmac-sha256:e4c3319a57b735e10df4677193bd7a969ee757a97ebf890bbaebb0694f6a6cd5",
+      "display_name":"token",
+      "policies":[
+         "root"
+      ],
+      "token_policies":[
+         "root"
+      ],
+      "policy_results":{
+         "allowed":true,
+         "granting_policies":[
+            {
+               "name":"root",
+               "namespace_id":"root",
+               "type":"acl"
+            }
+         ]
+      },
+      "token_type":"service",
+      "token_issue_time":"2022-09-09T22:01:47Z"
+   },
+   "request":{
+      "id":"50f70b23-3f2a-4857-5a8b-0c45982be491",
+      "client_id":"0DHqvq2D77kL2/JTPSZkTMJbkFVmUu0TzMi0jiXcFy8=",
+      "operation":"create",
+      "mount_type":"kv",
+      "mount_accessor":"kv_bf4b26f3",
+      "client_token":"hmac-sha256:2d8d7aeded539495117218a14ef9ae9f0f525eb2f6078be25dfe8e329ceb3d3f",
+      "client_token_accessor":"hmac-sha256:e4c3319a57b735e10df4677193bd7a969ee757a97ebf890bbaebb0694f6a6cd5",
+      "namespace":{
+         "id":"root"
+      },
+      "path":"secret/data/vault-demo-app,dev",
+      "data":{
+         "data":{
+            "okta.oauth2.clientId":"hmac-sha256:6709ec7e5682f16893369660844392809842ee1d670697fb034c3db17cfdc3d4",
+            "okta.oauth2.clientSecret":"hmac-sha256:c3db998af7b44aeaabf3da71c396e8bf86119fd6c36e66a41de67e91f71b116f",
+            "okta.oauth2.issuer":"hmac-sha256:70502e5cd54d12004df6836d0ff8d045515e4207607d984da14c0aaca5400c0a"
+         },
+         "options":{
+
+         }
+      },
+      "remote_address":"127.0.0.1",
+      "remote_port":58914
+   }
+}
+```
+
+Let's assume you don't want to configure the root token in the `vault-demo-app`. You can instead create a policy granting read permissions on the path where the secrets were stored. Go to the Vault Web UI at `http://localhost:8200` and log in with the root token (`00000000-0000-0000-0000-000000000000`).
+
+{% img blog/spring-vault/vault-web-ui.png alt:"Vault web UI" width:"800" %}{: .center-image }
```

Comment:
The text in this image is kinda small. Can you please zoom in by bumping the fonts and update?

Reply:
Updated vault screenshots.